src/ClientData/ModuleManager.cpp

```diff
@@ -67,6 +68,40 @@ std::vector<ModuleData*> ModuleManager::AddOrUpdateNotLoadedModules(
   return not_updated_modules;
 }
 
+const ModuleData* ModuleManager::GetModuleByModuleInMemoryAndAbsoluteAddress(
+    const ModuleInMemory& module_in_memory, uint64_t absolute_address) const {
+  absl::MutexLock lock(&mutex_);
+  auto it =
+      module_map_.find(std::make_pair(module_in_memory.file_path(), module_in_memory.build_id()));
+  if (it == module_map_.end()) return nullptr;
+
+  // The valid absolute address should be >=
+  // module_base_address + (executable_segment_offset % kPageSize)
```

Comment:
So the `% kPageSize` was for the case when `executable_segment_offset` is not a multiple of the page size, but `module_base_address` always has to be? I'm wondering if this is clear enough, because I couldn't really find an explanation in the existing code.

Reply:
I just checked and we have CHECKs in Address.cpp to make sure base address in always page aligned